File: problems/cpp/reverse-vowels-of-a-string.cpp
class Solution {
public:
    bool isVowel(char c){
        c = tolower(c);
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }
    string reverseVowels(string s) {
        vector<int>pos;
        for(int i = 0; i < s.length(); i++){
            if(isVowel(s[i])){
                pos.push_back(i);
            }
        }
        for(int i = 0; i < pos.size() / 2; i++){
            swap(s[pos[i]], s[pos[pos.size() - i - 1]]);
        }
        return s;
    }
};

File: problems/python/reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        s = list(s)
        def isVowel(c):
            return c.lower() in ['a', 'e', 'i', 'o', 'u']
        vowels = list(filter(isVowel, s))
        ind = len(vowels) - 1
        for i in range(len(s)):
            if isVowel(s[i]):
                s[i] = vowels[ind]
                ind -= 1
        return '' . join(s)
